tests/mr/host/mr_ref_tests.cpp

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2019-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../../byte_literals.hpp"
+
+#include <rmm/detail/aligned.hpp>
+#include <rmm/mr/host/host_memory_resource.hpp>
+#include <rmm/mr/host/new_delete_resource.hpp>
+#include <rmm/mr/host/pinned_memory_resource.hpp>
+
+#include <cuda_runtime_api.h>
+
+#include <cuda/memory_resource>
+
+#include <gtest/gtest.h>
+
+#include <cstddef>
+#include <deque>
+#include <random>
+
+namespace rmm::test {
+namespace {
+inline bool is_aligned(void* ptr, std::size_t alignment = alignof(std::max_align_t))
+{
+  return rmm::detail::is_pointer_aligned(ptr, alignment);
+}
+
+// Returns true if a pointer points to a device memory or managed memory allocation.
+inline bool is_device_memory(void* ptr)
+{
+  cudaPointerAttributes attributes{};
+  if (cudaSuccess != cudaPointerGetAttributes(&attributes, ptr)) { return false; }
+#if CUDART_VERSION < 10000  // memoryType is deprecated in CUDA 10
+  return attributes.memoryType == cudaMemoryTypeDevice;
+#else
+  return (attributes.type == cudaMemoryTypeDevice) or (attributes.type == cudaMemoryTypeManaged);
+#endif
```

Comment:
Can we drop code meant to support anything below CUDA 11?
```suggestion
  return (attributes.type == cudaMemoryTypeDevice) or (attributes.type == cudaMemoryTypeManaged);
```